homeassistant/components/media_player/cast.py

```diff
@@ -188,21 +190,18 @@ def async_cast_discovered(chromecast):
         if cast_device is not None:
             hass.data[KNOWN_CHROMECASTS_KEY][key] = chromecast
             async_add_devices([cast_device])
-        else:
-            # Some other entity already has this chromecast, throw this one
-            # away.
-            chromecast.disconnect(blocking=False)
 
 
 class CastDevice(MediaPlayerDevice):
     """Representation of a Cast device on the network."""
 
-    def __init__(self, chromecast):
+    def __init__(self, hass, chromecast):
```

Comment:
You shouldn't pass the `hass` object into your constructor, I think. `self.hass` will be set by `add_devices`, just use that.

Reply:
Ah yes that's right - came up during adding the test but I think that was because I'm not using `async_setup_component`